3007 - Added new uplift colors

tmcconechy opened the pull request:
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Updated design system tokens to new colors for uplift and did a pass on all three theme variants. This impacts and improves many internal colors in components and charts. Also updated the look and feel of the colors page.

**Related github/jira issue (required)**:
Fixes #3007

**Steps necessary to review your pull request (required)**:
*  pull this branch
*  `npm i` for new tokens
*  restart demo server
Kitchen Sink https://3007-new-uplift-colors-enterprise.demo.design.infor.com/?theme=uplift&variant=light
* Uplift Colors / Color Palette Page Design http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/colors/example-index.html?theme=uplift
* Header Colors http://3007-new-uplift-colors-enterprise.demo.design.infor.com//patterns/builder-wizard.html?theme=uplift
* Button Colors http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/button/example-index.html?theme=uplift
                - Not much difference in tertiary hover state so made some changes
                - Not sure about some disabled buttons, made some minor adjustments
* Menu Buttons http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/menubutton/example-index.html?theme=uplift
* Context Menus http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/contextmenu/example-index.html?theme=uplift
* Inputs http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/input/example-index.html?theme=uplift
            - Do the “muted” labels look ok?
            - Check readonly/disabled states and borders and placeholders?
* Autocomplete http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/autocomplete/example-index.html?theme=uplift
* Datepicker http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/datepicker/example-index.html?theme=uplift (WIP needs revision?)
* Switch http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/switch/example-index.html?theme=uplift
* Editor http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/editor/example-index.html?theme=uplift
* Badges http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/badges/example-index.html?theme=uplift
* Chart (Bullet) http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/bullet/example-index.html?theme=uplift
* Chart (Sparkline) http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/sparkline/example-index.html?theme=uplift&variant=light
* Chart (bar) http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/bar-grouped/example-index.html?theme=uplift&variant=light
* Chart (area) http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/area/example-index.html?theme=uplift&variant=light
* Chart (Donut) http://3007-new-uplift-colors-enterprise.demo.design.infor.com//components/donut/example-index.html?theme=uplift&variant=light (edited) 

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

tmcconechy commented:
@elizabethhartley - please let me know if you have any comments

EdwardCoyle commented:
Here's some things I found when reviewing:

**Header/Theme Switcher** - doesn’t always set the colors correctly:
* header colors on Uplift “default” are different from Uplift “azure”
![Screen Shot 2020-01-07 at 9 42 30 AM](https://user-images.githubusercontent.com/3249601/71905423-1c3d5d00-3136-11ea-885d-60e8f80f9a9a.png)
![Screen Shot 2020-01-07 at 9 42 45 AM](https://user-images.githubusercontent.com/3249601/71905424-1c3d5d00-3136-11ea-8a19-f8865b0b14e8.png)

* Some font colors set to black against headers where it should probably be white
![Screen Shot 2020-01-07 at 9 43 12 AM](https://user-images.githubusercontent.com/3249601/71905473-34ad7780-3136-11ea-9775-88780ed08130.png)

**Autocomplete (Dark)**
* “highlight” color (seen when hovering/selecting with keyboard) is pretty close to the background color
![Screen Shot 2020-01-07 at 10 14 07 AM](https://user-images.githubusercontent.com/3249601/71905813-e482e500-3136-11ea-9ce2-0a36ef9fa23b.png)

**Datepicker**
* Modal Button focus states inside the picker are differently-sized (the submit button pops out, but the clear button doesn’t)
* Regarding your note about WIP: it actually doesn't look bad to me.
![Screen Shot 2020-01-07 at 9 53 28 AM](https://user-images.githubusercontent.com/3249601/71905499-455ded80-3136-11ea-96ce-9fc36361a687.png)
![Screen Shot 2020-01-07 at 9 53 36 AM](https://user-images.githubusercontent.com/3249601/71905500-455ded80-3136-11ea-82a4-305244103441.png)

* I believe the Datepicker configured with a legend is using Soho colors against the Uplift theme, or they are just hardcoded to one color palette (specifically I noticed the emerald and azure don't match up):
![Screen Shot 2020-01-07 at 10 14 46 AM](https://user-images.githubusercontent.com/3249601/71905737-c2896280-3136-11ea-945d-1e67e98224e0.png)

**Editor**
* Did the design for the popupmenus switch back to the standard design with an arrow?  If not, this branch might need an update from 4.24.x
![Screen Shot 2020-01-07 at 10 16 41 AM](https://user-images.githubusercontent.com/3249601/71905792-da60e680-3136-11ea-81d2-266dd883a8f6.png)

**Input fields**
* opinion: I actually like the muted colors for the labels a lot more than what we previously had.

**Sparkline Chart**
* there's something funky about one of the sparklines on this sample when using the Dark/Contrast variants:
![Screen Shot 2020-01-07 at 10 09 16 AM](https://user-images.githubusercontent.com/3249601/71905869-fc5a6900-3136-11ea-8ba6-16effdc56519.png)
![Screen Shot 2020-01-07 at 10 10 08 AM](https://user-images.githubusercontent.com/3249601/71905870-fc5a6900-3136-11ea-9e9a-7c8d98172fb8.png)

tmcconechy commented:
@EdwardCoyle I fixed all the points you noted. I didnt actually see point two of it changing black text. I suspect when i adjusted the colors on point 1 i fixed it? If not let me know
